packages/badge-service/badgeGenerator/badgeGenerator.ts

```diff
@@ -0,0 +1,74 @@
+const sha512 = require('js-sha512');
+import { MutationScoreMapper, MutationScore } from 'stryker-dashboard-data-access';
+import * as httpHelpers from '../helpers/httpHelpers';
+
+export async function run(context: any, req: any) {
+    let statusCode = 400;
+    context.res = {
+        status: statusCode,
+    };
+
+    const scoreRepo = new MutationScoreMapper();
+
+    let rowKey = context.bindingData.repo;
+
+    if (context.bindingData.branch) {
+        rowKey += `/${context.bindingData.branch}`;
+    }
+
+    try {
+        const mutationScore = await scoreRepo.selectSingleEntity(`${context.bindingData.provider}/${context.bindingData.owner}`, rowKey);
+        if (mutationScore) {
+            const scoreColor = determineColor(mutationScore.score);
+
+            // Retrieve badge
+            const badge = await retrieveBadge(scoreColor, mutationScore.score);
+
+            if(!badge) {
+                throw 'no valid image found';
+            }
+
+            statusCode = 200;
+
+            context.res = {
+                status: statusCode,
+                headers: {
+                    'Cache-Control': 'no-cache, no-store, must-revalidate',
+                    'Pragma': 'no-cache'
+                },
+                body: badge
+            };
+        }
+    }
+    catch (exception) {
+        statusCode = 500;
```

Comment:
It's probably better to log the error and return https://img.shields.io/badge/mutation%20score-unknown-lightgrey.svg, no?

Reply:
hmm, good one, I'll make that change

Reply:
I've made the change 👍